### PR TITLE
Implement internal/net/control test

### PR DESCRIPTION
**Pull request by kevindiu**
Signed-off-by: kevindiu <kevindiujp@gmail.com>

### Description:

This PR implements the test for internal/net/control.

### Environment:

- Go Version: 1.17.2
- Docker Version: 20.10.8
- Kubernetes Version: 1.22.0
- NGT Version: 1.13.7

### Types of changes:

- [ ] Bug fix [type/bug]
- [ ] New feature [type/feature]
- [x] Add tests [type/test]
- [ ] Security related changes [type/security]
- [ ] Add documents [type/documentation]
- [ ] Refactoring [type/refactoring]
- [ ] Update dependencies [type/dependency]
- [ ] Update benchmarks and performances [type/bench]
- [ ] Update CI [type/ci]

### Changes to Core Features:

- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [x] Have you written new tests for your core changes, as applicable?
- [x] Have you successfully ran tests with your changes locally?

### Checklist:

- [x] I have read the [CONTRIBUTING](https://github.com/vdaas/vald/blob/master/CONTRIBUTING.md) document.
- [x] I have checked open [Pull Requests](https://github.com/vdaas/vald/pulls) for the similar feature or fixes?
- [x] I have added tests and benchmarks to cover my changes.
- [x] I have ensured all new and existing tests passed.
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: internal/net/control/control.go [hunk 1]
```diff
@@ -24,6 +24,7 @@ import (
 	"github.com/vdaas/vald/internal/log"
 )
 
+// SocketController represent the socket listener controller.
```
Comment (kmrmt) [inline internal/net/control/control.go:27]: I think that "represents" is more correct representation grammatically.

File: internal/net/control/control.go [hunk 2]
```diff
@@ -41,6 +42,7 @@ type control struct {
 	keepAlive                int
 }
 
+// SockerFlag represent the flag to enable specific feature for the socket listener.
```
Comment (kmrmt) [inline internal/net/control/control.go:45]: I think that "represents" is more correct representation grammatically.

File: internal/net/control/control_test.go [hunk 3]
```diff
@@ -198,31 +191,42 @@ func Test_isTCP(t *testing.T) {
...
+			name: "return true when network is tcp6",
+			args: args{
+				network: "tcp6",
+			},
+			want: want{
+				want: true,
+			},
+		},
+		{
+			name: "return true when network not tcp",
```
Comment (kmrmt) [inline internal/net/control/control_test.go:222]: return false?

File: internal/net/control/control.go [hunk 4]
```diff
@@ -55,6 +57,7 @@ const (
 	IPRecoverDestinationAddr
 )
 
+// New return the socket controller.
```
Comment (kmrmt) [inline internal/net/control/control.go:60]: I think that "returns" is more correct representation grammatically.

File: internal/net/control/control_test.go (hunk 3 above)
Comment (kevindiu) [reply internal/net/control/control_test.go:222]:
```suggestion
			name: "return false when network is not tcp",
```

File: internal/net/control/control.go (hunk 4 above)
Comment (kevindiu) [reply internal/net/control/control.go:60]:
```suggestion
// New returns the socket controller.
```

File: internal/net/control/control.go (hunk 2 above)
Comment (kevindiu) [reply internal/net/control/control.go:45]:
```suggestion
// SocketFlag represents the flag to enable specific feature for the socket listener.
```

File: internal/net/control/control.go (hunk 1 above)
Comment (kevindiu) [reply internal/net/control/control.go:27]:
```suggestion
// SocketController represents the socket listener controller.
```

File: internal/net/control/control_test.go [hunk 5]
```diff
@@ -359,3 +327,183 @@ func Test_control_GetControl(t *testing.T) {
...
+				ipTransparent:            test.fields.ipTransparent,
+				ipRecoverDestinationAddr: test.fields.ipRecoverDestinationAddr,
+				keepAlive:                test.fields.keepAlive,
+			}
+
+			err := ctrl.controlFunc(test.args.network, test.args.address, test.args.c)
+			if err := test.checkFunc(test.want, err); err != nil {
+				tt.Errorf("error = %v", err)
+			}
+
```
[the remark this answers is not in the document]
Comment (kevindiu) [reply internal/net/control/control_test.go:506]:
```suggestion
```

**Review by kmrmt — DISMISSED** [review]

**Review by vankichi — DISMISSED** [review]

**Pull request merged**